src/main/ApplicationImpl.cpp

```diff
@@ -112,6 +113,7 @@ ApplicationImpl::initialize()
     mHerderPersistence = HerderPersistence::create(*this);
     mBucketManager = BucketManager::create(*this);
     mCatchupManager = CatchupManager::create(*this);
+    mHistoryArchiveMonitor = make_unique<HistoryArchiveMonitor>();
```

Comment:
Looks to me that `mHistoryArchiveMonitor`  should live under `HistoryManager` and not as a child of the application as history archives are an implementation detail of how history works